### PR TITLE
[iOS] ✨ Add PagingScrollView

### PR DESCRIPTION
**Pull request by tomas-bat**
# :pencil: Description
- This PR adds a `PagingScrollView` component, which can be useful for handling paging.

# :bulb: What’s new?
- The `PagingScrollView` is a scroll view with the ability to call a given function when the scroll offset reaches a certain percentage. This function serves as a trigger for fetching a next page.

# :no_mouth: What’s missing?
- There are no appropriate views in DevStack where the `PagingScrollView` could be used, because it requires the data to come in a paging type. A usage example can be found in Můj Up, for example [here](https://github.com/MateeDevs/updigital-ios/blob/develop/PresentationLayer/Sources/PresentationLayer/App/Home/UserProfile/MyProfile/PersonalProfileBadges/PersonalProfileAllBadges/PersonalProfileAllBadgesView.swift).

# :books: References
- https://github.com/MateeDevs/updigital-ios/pull/1071

### CODE REVIEW

File: ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/ScrollView/PagingScrollView.swift [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    ///  - isFetchingMore: Whether data is currently being fetched – this prevents from calling `touchedBottomAction` again
+    ///  - triggerPoint: At which point the `touchedBottomAction` should be triggered, 0.0 would be at the very top, 1.0 at the very bottom
+    ///  - touchedBottomAction: The function that should be called when the trigger point is reached
+    ///  - content: The content for the scroll view
+    init(
+        _ axes: SwiftUI.Axis.Set = .vertical,
+        showsIndicators: Bool = true,
+        triggerPoint: Double = 1.0,
+        isFetchingMore: Bool,
+        touchedBottomAction: (() -> Void)?,
```
Comment (DavidKadlcek) [inline ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/ScrollView/PagingScrollView.swift]: I would prefer to put there `= nil`

Comment (DavidKadlcek) [reply ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/ScrollView/PagingScrollView.swift]: Also, rename it to `triggerAction` or something like that

File: ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/ScrollView/PagingScrollView.swift [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    ) {
+        self.axes = axes
+        self.showsIndicators = showsIndicators
+        self.triggerPoint = triggerPoint
+        self.isFetchingMore = isFetchingMore
+        self.touchedBottomAction = touchedBottomAction
+        self.content = content()
+    }
+    
+    var body: some View {
```
Comment (DavidKadlcek) [inline ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/ScrollView/PagingScrollView.swift]: public

File: ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/ScrollView/PagingScrollView.swift [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    private let content: Content
+    
+    /// - parameters:
+    ///  - axes: The scroll view axes
+    ///  - showIndicators: Whether the scroll indicators are shown
+    ///  - isFetchingMore: Whether data is currently being fetched – this prevents from calling `touchedBottomAction` again
+    ///  - triggerPoint: At which point the `touchedBottomAction` should be triggered, 0.0 would be at the very top, 1.0 at the very bottom
+    ///  - touchedBottomAction: The function that should be called when the trigger point is reached
+    ///  - content: The content for the scroll view
+    init(
```
Comment (DavidKadlcek) [inline ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/ScrollView/PagingScrollView.swift]: public

File: ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/ScrollView/PagingScrollView.swift [hunk 4]
```diff
@@ -0,0 +1,95 @@
+//
+//  Created by Tomáš Batěk on 10.07.2023
+//  Copyright © 2023 Matee. All rights reserved.
+//
+
+import SwiftUI
+
+/// A Scroll View with the ability to call a certain functionality at a given percentage of scroll
+/// - note: Taken from https://stackoverflow.com/questions/68681075/how-do-i-detect-when-user-has-reached-the-bottom-of-the-scrollview
+struct PagingScrollView<Content: View>: View {
```
Comment (DavidKadlcek) [inline ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/ScrollView/PagingScrollView.swift]: public

**Comment by pchmelar** [comment]
`GET /users` has paging, can it be used for example of usage? 👀 
https://devstack-server-production.up.railway.app/apidoc.html#request-user-get-user

**Comment by tomas-bat** [comment]
> `GET /users` has paging, can it be used for example of usage? 👀
> https://devstack-server-production.up.railway.app/apidoc.html#request-user-get-user

While implementing it there, I decided to add the `PagingHandler` from Můj Up as well, so this will go through more changes 😅 Marking as WIP for now.

File: ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/ScrollView/PagingScrollView.swift (hunk 1 above)
Comment (tomas-bat) [reply ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/ScrollView/PagingScrollView.swift]: Actually, I removed the optional wrap.. It doesn't really make sense to use `PagingScrollView` without this action 😅

File: ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/ScrollView/PagingScrollView.swift (hunk 2 above)
Comment (tomas-bat) [reply ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/ScrollView/PagingScrollView.swift]: Done

File: ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/ScrollView/PagingScrollView.swift (hunk 3 above)
Comment (tomas-bat) [reply ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/ScrollView/PagingScrollView.swift]: Done

File: ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/ScrollView/PagingScrollView.swift (hunk 4 above)
Comment (tomas-bat) [reply ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/ScrollView/PagingScrollView.swift]: Done

**Review by DavidKadlcek — APPROVED** [review]

**Pull request merged**